src/state/user/hooks.tsx

```diff
@@ -499,3 +476,35 @@ export const useUserFavoriteTokens = (chainId: ChainId | undefined) => {
 
   return { favoriteTokens, toggleFavoriteToken }
 }
+
+export const useAllDexes = () => {
+  const { chainId } = useActiveWeb3React()
+  const dexes = useSelector<AppState, AppState['user']['allDexes']>(state => state.user.allDexes)
+
+  return useMemo(() => {
+    if (!chainId) return []
+    return dexes[chainId]
+  }, [chainId, dexes])
+}
+
+export const useExcludeDexes = (): [string[], (value: string[]) => void] => {
+  const { chainId } = useActiveWeb3React()
+  const dispatch = useAppDispatch()
+  const excludeDexes = useSelector<AppState, AppState['user']['excludeDexes']>(state => state.user.excludeDexes)
+
+  const excludeDexesByChainId: string[] = useMemo(() => {
+    if (!chainId) return []
+    return excludeDexes[chainId] || []
+  }, [chainId, excludeDexes])
+
+  const setExcludeDexes = useCallback(
+    (dexes: string[]) => {
+      console.log(dexes)
```

Comment:
This should be removed

Reply:
Maybe we should add a new lint rule for console.log

Reply:
removed.

Reply:
yeah. should add warning for it